libfoundation/src/foundation-string.cpp

```diff
@@ -6963,6 +6963,60 @@ bool MCStringGetNumericValue(MCStringRef self, double &r_value)
         return false;
 }
 
+MC_DLLEXPORT
+bool MCStringNormalizeLineEndings(MCStringRef p_input, 
+                                  MCStringLineEndingStyle p_to_style, 
+                                  MCStringRef& r_output, 
+                                  MCStringLineEndingStyle* r_original_style)
+{
+    uindex_t t_firstLF = 0;
+    uindex_t t_firstCR = 0;
+    MCStringLineEndingStyle t_original_style = kMCStringLineEndingStyleLF;
+    MCStringRef t_mutable_input;
+    
+    if (MCStringFirstIndexOfChar(p_input, '\r', 0, kMCStringOptionCompareExact, t_firstCR))
```

Comment:
I suggest variable definitions should be just before the block which first uses them, with ones which are needed outside the block first:
```
MCStringLineEndingStyle t_original_style...
uindex_t t_first_lf ...
uindex_t t_first_cr
```